lib/airbrake-ruby/filter_chain.rb

```diff
@@ -67,7 +68,8 @@ def root_directory_filter(root_directory)
       proc do |notice|
         notice[:errors].each do |error|
           error[:backtrace].each do |frame|
-            frame[:file].sub!(/\A#{root_directory}/, '[PROJECT_ROOT]'.freeze)
+            next unless (file = frame[:file])
+            file.sub!(/\A#{root_directory}/, '[PROJECT_ROOT]'.freeze)
           end
```

Comment:
I don't think it would be possible to do that without creating messy code. This particular filter executes only if the `root_direction` option is defined. This proc doesn't know about the config and I wouldn't want to pass it just for 1 check. They look similar only on the surface.